### ISSUE TITLE
Restructure http/browser_client.dart in http package to avoid dart:mirrors usage (?)

### ISSUE THREAD

**Issue by DartBot** [dart-lang/http#25]
Blocked by https://github.com/dart-lang/sdk/issues/24581

**Comment by DartBot** [comment]
<a href="https://github.com/Scorpiion"><img src="https://avatars.githubusercontent.com/u/1047298?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [Scorpiion](https://github.com/Scorpiion)**

----

To be clear, a simple Hello World app importing package:http/browser_client.dart gives this:

\* Your app imports dart:mirrors via:
\*   main.dart => package:http => dart:mirrors

**Comment by DartBot** [comment]
<a href="https://github.com/lrhn"><img src="https://avatars.githubusercontent.com/u/4865287?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [lrhn](https://github.com/lrhn)**

----

*Removed [Type-Defect](../labels/Type-Defect) label.*
*Added [Type-Enhancement](../labels/Type-Enhancement), [Pkg-Http](../labels/Pkg-Http), [Area-Pkg](../labels/Area-Pkg), [Triaged](../labels/Triaged) labels.*

**Comment by DartBot** [comment]
<a href="https://github.com/nex3"><img src="https://avatars.githubusercontent.com/u/188?v=3" align="left" width="48" height="48"hspace="10"></img></a> **Comment by [nex3](https://github.com/nex3)**

----

The warning sucks, but in practice the usage of mirrors in the http package is harmless. It's only used to determine that dart:io is unavailable, and adds essentially no size overhead to the compiled output.

I'd love to remove mirrors from the package, but right now the language just doesn't support it while maintaining the same API.
______
*Marked this as being blocked by dart-lang/sdk#6943.*

**Comment by polux** [comment]
The usage of mirrors makes it impossible to use this lib with flutter (which disables mirrors).

**Comment by sethladd** [comment]
Yes. We had to fork http here: https://github.com/flutter/flutter/blob/master/packages/flutter/lib/http.dart

**Comment by nex3** [comment]
Disabling mirrors makes sense, but making programs statically invalid just by importing `dart:mirrors` seems like overkill. Why not just have `createMirrorSystem()` throw an error? Then you could use HTTP and any other libraries that use similar workarounds.

**Comment by nex3** [comment]
I'm merging this as a duplicate of #1.
